server/models.py

```diff
@@ -13,23 +14,26 @@
 import playhouse.postgres_ext as pw_postgres
 
 
-HASHING_ALGORITHM = hashlib.sha3_512
+def hash_password(password: str) -> str:
+    """Hash a password."""
+    salt = os.urandom(32)
+    key = hashlib.pbkdf2_hmac('sha3-512', password.encode(), salt, 100_000)
+    return salt + key
 
-db = pw_postgres.PostgresqlExtDatabase(
-    config.DB_NAME, user=config.DB_USER, password=config.DB_PASSWORD
-)
 
+def check_password(password: str, hashed: str) -> bool:
+    """Check a password against a hash."""
+    salt = hashed[:32]
+    key = hashed[32:]
+    attempt_key = hashlib.pbkdf2_hmac(
+        'sha3-512', password.encode(), salt, 100_000
+    )
+    return key == attempt_key
```

Comment:
This is also insecure. As much as timing attacks are unlikely it's better to be safe, and use the `hmac.compare_digits` function, which is cryptographically secure. See [Coda Hale’s blog post about the timing attacks on KeyCzar and Java’s `MessageDigest.isEqual()`](https://codahale.com/a-lesson-in-timing-attacks/).